File: eshop-parent/eshop-goods-interface/src/main/java/com/eshop/goods/service/BrandService.java
package com.eshop.goods.service;

import java.util.List;

import com.eshop.common.PageResult;
import com.eshop.pojo.TbBrand;

/**
 * 品牌接口
 * @author yzm
 *
 */
public interface BrandService {
	
	public List<TbBrand> findAll();   
	
	/**
	 * 品牌分页
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	/**
	 * 增加品牌
	 * @param brand
	 */
	public void add(TbBrand brand);
	
	/**
	 * 根据Id查询实体
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);
	
	/**
	 * 根据查询的实体进行修改
	 * @param brand
	 * @return
	 */
	public void update(TbBrand brand);
	
	/**
	 * 删除
	 * @param id
	 */
	public void delete(Long[] ids);
	
	/**
	 * 根据查询条件查询
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbBrand brand,int pageNum,int pageSize);
}

File: eshop-parent/eshop-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService',function($http){
    	//读取列表数据绑定到表单中
    	this.findAll=function(){
    		return $http.get('../brand/findAll.do');		
    	}
    	//分页 
    	this.findPage=function(page,size){
    		return $http.get('../brand/findPage.do?pageNum='+page+'&pageSize='+size);		
    	}
    	
    	//查询实体
    	this.findOne=function(id){
    		return $http.get('../brand/findOne.do?id=' + id);		
    	}    
    	
    	//增加 
    	this.add=function(entity){
    		return $http.post('../brand/add.do',entity);		
    	}
    	
    	//更新
    	this.update=function(entity){
    		return $http.post('../brand/update.do',entity);		
    	}
    	
    	//删除
    	this.del=function(ids){
    		return $http.get('../brand/delete.do?ids=' + ids);		
    	}

    	//搜索
    	this.search=function(page,size,searchEntity){
    		return $http.post('../brand/search.do?page='+page +'&size='+size,searchEntity);		
    	}
	}); 